modules/miner/cpuminer.go

```diff
@@ -2,11 +2,18 @@ package miner
 
 import (
 	"time"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
 // threadedMine starts a gothread that does CPU mining. threadedMine is the
 // only function that should be setting the mining flag to true.
 func (m *Miner) threadedMine() {
+	if err := m.tg.Add(); err != nil {
+		return
+	}
+	defer m.tg.Done()
+
```

Comment:
@lukechampine not quite correct.

`threaded` indicates that this function should always be called as a goroutine (e.g. `go threadedMine()`), which means that it's possible the miner closes between the time this function is called and the time that this function executes.

So, the first lines of code here should actually be to check whether the miner is closed or not. @mnsl it's not a critical error if this function gets called after the miner is closed - it's possible during normal use for this function to be called, and then for the miner to close, and then for this function to execute. So if the miner is closed when this function starts, you should just `return`, silently refusing to start the miner.